miru: init at 4.5.10

d4ilyrun opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

onny commented:
Please rebase

abathur commented:
@d4ilyrun Sorry this has languished without getting a proper review.

We can try to get it moving, but unfortunately I'll have to ask you to jump through a few more hoops...:
- You'll need to rebase against nixpkgs master again. It looks like both pkgs/top-level/all-packages.nix and maintainers/maintainer-list.nix have merge conflicts to fix.
- Please update the PR title to match the new version target (it still indicates 2.2.2, but the commit indicates 3.10.6).
- The CI report from your last push indicates at least 2 problems:
    - Your maintainer entry isn't in the right order alphabetically. It looks like you appended it to the end, but it'll need to go in the right spot. At the time I'm writing this, I imagine that means it needs to go between `cyplo` and `dadada`.
    - When you updated the PR to target version 3.10.6, you didn't update the hash/sha256. You'll need to update it to match the new source (and make sure the project still builds/works without needing additional changes).
    
        (You likely didn't notice this problem on your end since you'd already built the 2.2.2 version on your system. Nix uses this hash to look for an existing copy in the store--it won't rebuild if you change the version but leave the hash alone. If you empty the hash string and try to build it, Nix will pull the source again and let you know what the new hash is.)
        
- @onny marked your PR as a draft when he asked you to rebase, which lets reviewers know that the PR needs some work or changes before it's reviewable. Once you make the changes on my list and re-push, please also mark it as ready for review!

(Fair warning: I'm just asking for the things I'm pretty sure you'll need to get this ready for review. Reviewers will likely request at least one more round of changes, if not a few.)

abathur commented:
I noticed there's another draft PR open to add this package. If you decide to put additional effort into this one, check up on its progress first:
- #238429

If it bogs down, you can also use the review feedback it received as a reference to help handle some issues before review.

d4ilyrun commented:
Hello @abathur ! 
Sorry I took so long, I just rebased my branch and followed your advice, it should be good for another review now 😄

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2966

onemoresuza commented:
There's a new convention for packages path: [by-name](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md).

In this case, `pkgs/applications/video/miru/default.nix` should be moved to `pkgs/by-name/mi/miru/package.nix`.

abathur commented:
I'm not a decider on this and don't know if we've concluded all new packages _must_ follow the by-name convention--and it's moot if the author eagerly adopts the convention--but I do want to ask anyone making a review/merge call on this point to weigh the risk of bogging down this long-deferred PR over something that wasn't a norm when it was opened and can almost certainly be handled during the eventual auto-migration.

onemoresuza commented:
Fair point, @abathur.

onemoresuza commented:
# Reviewed points

Result of `nixpkgs-review pr 172872` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>miru</li>
  </ul>
</details>

- [X] package name fits guidelines
- [X] package version fits guidelines
- [X] package builds on x86_64-linux
- [X] executables tested on x86_64-linux
- [X] all depending packages build
- [X] `AppImage` application states its `sourceProvenance` as `binaryNativeCode`

# Possible improvements

- The `mainProgram` meta attribute does not match the binary under `$out/bin`:
  instead of `miru`, there's a `miru-4.5.5`. An explicit copy of the
  binary to `$out/bin/miru` should fix it.

d4ilyrun commented:
Oh sorry I did not see there was a new way of adding packages.
 I updated everything to follow the new convention, and renamed the resulting binary's name.

onemoresuza reviewed (approved):
Good job, @d4ilyrun. Just remember that before a commiter may merge this PR, the commits
ought to be squashed.

SuperSandro2000 commented on pkgs/by-name/mi/miru/package.nix:
```diff
@@ -0,0 +1,55 @@
+appimageTools.wrapType2 rec {
+  pname = "miru";
+  version = "4.5.5";
+  name = "${pname}-${version}";
```
We no longer need to generate a name variable since all tools accept pname version.

d4ilyrun replied:
I redefined it since a similar pattern appeared inside the url and install commands.  
Just removed it and squashed everything.

heisfer commented:
I think most of the stuff can be copy pasta from here https://github.com/heisfer/nixdot/blob/main/pkgs/miru/default.nix

AndersonTorres commented on pkgs/by-name/mi/miru/package.nix:
```diff
@@ -0,0 +1,54 @@
+    longDescription = ''
+      A pure JS BitTorrent streaming environment, with a built-in list manager.
+      Imagine qBit + Taiga + MPV, all in a single package, but streamed real-time.
+      Completly ad free with no tracking/data collection.
```
```suggestion
      Completely ad free with no tracking/data collection.
```

AndersonTorres commented on pkgs/by-name/mi/miru/package.nix:
```diff
@@ -0,0 +1,54 @@
+      instead of flat out closing MPV.
+    '';
+  };
+
```
```suggestion
```

AndersonTorres reviewed (changes requested):
Besides, LFTM

d4ilyrun committed:
maintainers: add d4ilyrun

d4ilyrun committed:
miru: init at 4.5.10

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1450
